src/libpriv/rpmostree-rpm-util.c

```diff
@@ -1455,3 +1455,19 @@ rpmostree_get_rojig_branch_pkg (DnfPackage *pkg)
                                  dnf_package_get_evr (pkg),
                                  dnf_package_get_arch (pkg));
 }
+
+GPtrArray*
+rpmostree_get_enabled_rpmmd_repos (DnfContext *dnfctx,
+                                   DnfRepoEnabled enablement)
+{
+  g_autoptr(GPtrArray) ret = g_ptr_array_new ();
+  GPtrArray *repos = dnf_context_get_repos (dnfctx);
```

Comment:
One side note...from the current libdnf source I'm not sure this order is stable.  They are pulled unordered from `g_dir_read_name()` and passed to `g_ptr_array_sort()` which is guaranteed stable, but we're not usually adding cost, so the array will remain unordered.

We should probably do something more like "sort by cost, then name" or something in libdnf?

(We want this to be stable or otherwise we'll be reordering things in the JSON)

Reply:
At least as far as the JSON output, we use a `BTreeMap` on the repo name, so the order in which we serialize should already be stable (learned that lesson from https://github.com/coreos/rpm-ostree/pull/1865).

Reply:
Ah right, OK.